### ISSUE TITLE
New option to leave browser open upon first error

### ISSUE THREAD

**Issue by binarykitchen** [webdriverio/webdriverio#2669]
Would be nice to have an option telling not to close browser when the first error has occured.

But close it when all pass.

**Comment by christian-bromann** [comment]
Can you please explain the feature request in more detail? In which scenario you would use this and how could it be useful. This doesn't tell me anything :-/

**Comment by binarykitchen** [comment]
ok. this scenario:
1. running a couple of end to end tests in one single run
2. at least one of them throws an error
3. i want to investigate and see the problem in the browser 
4. unfortunately it's already closed

hence i'd like a new option to keep the browser open only upon failing test cases.

not keen for screenshots here, want to inspect in the browser. thanks!

**Comment by MDMazzullo** [comment]
Something I've done to accomplish this is to add the following code in my afterStep hook (I'm using the cucumber framework).
```
if (stepResult.getStatus() === "failed") {
    browser.debug();
}
```
Works fine for me, and if the cucumber timeout is long enough it will hold the test in that step before failing it and closing out the browser.

**Comment by christian-bromann** [comment]
@binarykitchen I like the idea from @MDMazzullo. It will do exactly what you describe. Introducing a new option that closes the browser and looses the reference to the WebDriver session is kind of problematic.

**Comment by diachedelic** [comment]
For those not using the cucumber framework, you can use this hook:

```javascript
  after: function(result) {
    if (result === 1) {
      browser.debug()
    }
  },
```

**Comment by ElvenMonky** [comment]
@diachedelic Thanks, you solution is really helpful!

**Comment by louisremi** [comment]
I used the cucumber-boilerplate project, and added the following snippet to my `wdio.BUILD.conf.ts` file:
```js
buildConfig.afterStep = (step, scenario, result) => {
    if (result.error) {
        (browser as WebdriverIO.Browser).debug();
        process.exit(1);
    }
};
```

**Comment by christian-bromann** [comment]
@louisremi make sure using latest version of WebdriverIO to use `await browser.debug()` since everything is async now.
